python/src/aiconfig/editor/client/src/components/JSONRenderer.tsx

```diff
@@ -16,12 +16,27 @@ export default memo(function JSONRenderer({
   schema,
 }: Props) {
   const { readOnly } = useContext(AIConfigContext);
-  // Prism is a read only renderer.
-  return !onChange || readOnly ? (
-    <Prism language="json" styles={{ code: { textWrap: "pretty" } }}>
-      {JSON.stringify(content, null, 2)}
-    </Prism>
-  ) : (
-    <JSONEditor content={content} onChangeContent={onChange} schema={schema} />
-  );
+  let renderer = null;
+
+  if (!onChange || readOnly) {
+    if (content != null) {
+      // Prism is a read only renderer, but requires a string input. Just render
+      // null if the content is empty in readonly
```

Comment:
thanks for the comment, took me a while to understand the the only difference in this PR was L22 for the null-check

Reply:
Ya, just figured it was better to pull into if/else rather than another ternary :)